cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTable.java

```diff
@@ -367,6 +368,28 @@ public void deleteAll(Range range) throws InvalidFieldException {
     transactionContext.executeUpdate(builder.build());
   }
 
+  @Override
+  public void updateAll(Range keyRange, Collection<Field<?>> fields) throws InvalidFieldException, IOException {
+    fieldValidator.validatePrimaryKeys(keyRange.getBegin(), true);
```

Comment:
I agree the `keyRange` must be primary keys in order. Let's also specify it in the interface javaDoc 

Also pls check the passed in `fields` is a valid field (`fieldValidator::validateField`) but not a primary key.